Build Failures - test_stringified_slice_with_tz

As noted by @TomAugspurger [here](https://github.com/pandas-dev/pandas/pull/25491#issuecomment-468519258) looks like builds started failing today. I think this has to do with the upcoming DST transition.

cc @mroeschke 

```sh
=================================== FAILURES ===================================
_______________ TestDatetimeIndex.test_stringified_slice_with_tz _______________
[gw0] darwin -- Python 3.5.6 /Users/vsts/miniconda3/envs/pandas-dev/bin/python

self = <pandas.tests.indexes.datetimes.test_datetime.TestDatetimeIndex object at 0x13accba58>

    def test_stringified_slice_with_tz(self):
        # GH#2658
        import datetime
        start = datetime.datetime.now()
>       idx = date_range(start=start, freq="1d", periods=10, tz='US/Eastern')

pandas/tests/indexes/datetimes/test_datetime.py:105: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pandas/core/indexes/datetimes.py:1551: in date_range
    closed=closed, **kwargs)
pandas/core/arrays/datetimes.py:454: in _generate_range
    tz, ambiguous=ambiguous, nonexistent=nonexistent)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   raise pytz.NonExistentTimeError(stamp)
E   pytz.exceptions.NonExistentTimeError: 2019-03-10 02:01:53.911138

pandas/_libs/tslibs/conversion.pyx:1100: NonExistentTimeError
```

--- a/pandas/tests/indexes/datetimes/test_datetime.py
+++ b/pandas/tests/indexes/datetimes/test_datetime.py
@@ -100,8 +100,7 @@ class TestDatetimeIndex(object):
 
     def test_stringified_slice_with_tz(self):
         # GH#2658
-        import datetime
-        start = datetime.datetime.now()
+        start = '2013-01-07'
         idx = date_range(start=start, freq="1d", periods=10, tz='US/Eastern')
         df = DataFrame(lrange(10), index=idx)
         df["2013-01-14 23:44:34.437768-05:00":]  # no exception here

